app/utils/url/path.ts

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+export const ID_PATH_PATTERN = '[a-z0-9]{26}';
+export const PLUGIN_ID_PATH_PATTERN = '^[a-zA-Z0-9-_.]+$';
```

Comment:
Because of the `^` and `$` anchoring this will never match, I think? In either case it needs test coverage.